aidial_sdk/header_propagator.py

```diff
@@ -8,10 +8,11 @@
 from fastapi import FastAPI
 from requests import PreparedRequest
 from requests.sessions import Session
+from starlette.middleware import _MiddlewareClass
 from starlette.middleware.exceptions import ExceptionMiddleware
 
 
-class FastAPIMiddleware:
+class FastAPIMiddleware(_MiddlewareClass):
```

Comment:
Let's try to avoid using private classes: https://github.com/encode/starlette/blob/master/tests/middleware/test_middleware.py#L15

Reply:
<img width="458" alt="image" src="https://github.com/epam/ai-dial-sdk/assets/24621956/17d8dc34-e897-491c-9828-0fcfe1b3b275">

`add_middleware` expects a middleware with this type